src/main/java/org/jenkinsci/plugins/workflow/job/properties/DisableResumeJobProperty.java

```diff
@@ -0,0 +1,26 @@
+package org.jenkinsci.plugins.workflow.job.properties;
+
+import hudson.Extension;
+import jenkins.model.OptionalJobProperty;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Disables resuming a pipeline if the master restarts - the run will simply fail instead, just like a FreeStyle job.
```

Comment:
Should then be paired with a change to `workflow-durable-task-step` to make `sh`/etc. use synchronous mode.

Reply:
That's [JENKINS-47390](https://issues.jenkins-ci.org/browse/JENKINS-47390) -- a separate work item.